### PR TITLE
5.0 MCH first pass, with new data and lots of nuked modules

### PR DESCRIPTION
**Pull request by Toastdeib**
Sorry for size. :( Shouldn't be too bad tho, since half of it is just deletions or string updates.

### CODE REVIEW

File: src/parser/jobs/mch/modules/Heat.js [hunk 1]
```diff
@@ -7,244 +7,71 @@ import {ActionLink} from 'components/ui/DbLink'
...
-					this._setBarrelState(BARREL_STATE.GAUSS)
-				}
-			} else if (abilityId === ACTIONS.COOLDOWN.id) {
-				this._lockoutCooldowns++
-			} else if (abilityId === ACTIONS.GAUSS_BARREL.id) {
-				this._fixNullAssumption()
-			}
-			break
+		if (event.timestamp > (this._overheatWindows.current.start + OVERHEAT_DURATION_MILLIS)) {
+			this._finishOverheatWindow()
```
Comment (ackwell) [inline src/parser/jobs/mch/modules/Heat.js]: I'm guessing there's no OH status to track?

File: src/parser/jobs/mch/modules/Gauge.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+		if (gauge.current >= MAX_GAUGE) {
+			gauge.waste += (gauge.current - MAX_GAUGE)
+			gauge.current = MAX_GAUGE
+		}
+
+		this._pushToHistory(type)
+	}
+
+	_onHeatGcd() {
+		this._addGauge('heat', GCD_HEAT_GAIN)
```
Comment (ackwell) [inline src/parser/jobs/mch/modules/Gauge.js]: Feel free to use inline arrows for super short event handlers like this. Stuff like

`this.addHook('type', {filter:stuff}, () => this._addGauge('foo', bar))`

File: src/parser/jobs/mch/modules/Gauge.js [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+	_onQueen() {
+		this._gauge.battery.current = 0
+		this._pushToHistory('battery')
+	}
+
+	_onDeath() {
+		this._gauge.heat.current = 0
+		this._gauge.battery.current = 0
+		// TODO - Include this in the waste calculation?
```
Comment (ackwell) [inline src/parser/jobs/mch/modules/Gauge.js]: Personally, I don't include deaths in waste - the death is already reported as a suggestion.

File: src/parser/jobs/mch/modules/Gauge.js [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+			},
+			value: this._gauge.battery.waste,
+			why: <Trans id="mch.gauge.suggestions.battery-waste.why">
+				You lost {this._gauge.battery.waste} Battery to an overcapped gauge.
+			</Trans>,
+		}))
+	}
+
+	output() {
+		// TODO
```
Comment (ackwell) [inline src/parser/jobs/mch/modules/Gauge.js]: Still to do?

**Comment by ackwell** [comment]
Ergh, translation conflicts. Will merge once resolved.

File: src/parser/jobs/mch/modules/Heat.js (hunk 1 above)
Comment (Toastdeib) [reply src/parser/jobs/mch/modules/Heat.js]: Yeh, OH is still gauge only. It just shows an 8 ticking down on your heat bar when you pop Hypercharge.

File: src/parser/jobs/mch/modules/Gauge.js (hunk 4 above)
Comment (Toastdeib) [reply src/parser/jobs/mch/modules/Gauge.js]: Nope, I can kill that.

File: src/parser/jobs/mch/modules/Gauge.js (hunk 3 above)
Comment (Toastdeib) [reply src/parser/jobs/mch/modules/Gauge.js]: Yeh, I'll clean the TODO up.

File: src/parser/jobs/mch/modules/Gauge.js (hunk 2 above)
Comment (Toastdeib) [reply src/parser/jobs/mch/modules/Gauge.js]: Fair, I can tidy up this and `_onStabilizer()`.

**Pull request merged**